tools/utils/CLAUDE.md

```diff
@@ -0,0 +1,64 @@
+# CLAUDE.md - Agent Rules for tools/utils/
+
+Rules for AI agents contributing to IREE developer utilities.
+
+## Critical Rules
+
+1. **Never `print()` directly** - Use `console.error/warn/note/success` for messages
+2. **Never `sys.exit(N)`** - Use `return exit_codes.SUCCESS/ERROR/NOT_FOUND`
+3. **Never raw file writes** - Use `fs.safe_write_text()` for atomic UTF-8 writes
+4. **Always add common flags** - Call `cli.add_common_output_flags(parser)`
+5. **JSON purity** - Never mix `print()` with JSON output; use `console.print_json()`
+
+## Test Command
+
+```bash
+python -m unittest discover -s tools/utils -p "test_*.py" -v
+```
+
+**Expected output**: `Ran 356 tests ... OK`
```

Comment:
I've found exact numbers to throw off the bot sometimes, can we do something like this?
```suggestion
**Expected output**: `Ran {{[0-9]+}} tests ... OK`
```